app/src/main/java/com/pileproject/drive/util/bluetooth/RxBluetoothConnector.java

```diff
@@ -77,11 +81,47 @@ public void call(Subscriber<? super BluetoothSocket> subscriber) {
 
                     subscriber.onNext(socket);
                     subscriber.onCompleted();
-                } catch (IOException e) {
-                    subscriber.onError(e);
+                } catch (IOException firstCause) {
+
+                    try {
+                        BluetoothSocket socket = connectByRfcommSocket(bluetoothDevice);
+                        subscriber.onNext(socket);
```

Comment:
I see and agree to keep the current implementation. I think keeping its simplicity as an example is also helpful for other developers :)